### PR TITLE
Remove `__getitem__` code duplication in `gensim.models.phrases`

### LINKED ISSUE #2149: Duplication of code of phrases module
I was going to submit a minor pull request for the phrases module, but then I noticed that the `__getitem__` methods of the classes `Phrases` and `Phraser` are identical. Would it be a good idea to move this code to a higher level, e.g. the `PhrasesTransformation` class, or somehow merge the duplicated code?

### PR DESCRIPTION
**Pull request by jenishah**
Fix #2149 
wrote a single function `sentence2token` that can be used by` _getitem_` of Phrase and Phrases class.

### CODE REVIEW

File: gensim/models/phrases.py [hunk 1]
```diff
@@ -232,6 +232,42 @@ def load(cls, *args, **kwargs):
         return model
 
 
+def sentence2token(phrase_class, sentence):
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: "hide" function a bit, i.e. start the name with `_`

File: gensim/models/phrases.py [hunk 2]
```diff
@@ -232,6 +232,42 @@ def load(cls, *args, **kwargs):
...
+    """ returns tokens after from a sentence by joining the phrases with delimeter.
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: Please use numpy-style docstring (as almost everywhere in gensim), see
- https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard
- https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html

File: gensim/models/phrases.py [hunk 3]
```diff
@@ -232,6 +232,42 @@ def load(cls, *args, **kwargs):
...
+    Used by __getitem__ of Phraser and Phrases
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: should be a link, i.e.
```
 :meth:`~gensim.models.phrases.Phrases.__getitem__` 
```

File: gensim/models/phrases.py [hunk 4]
```diff
@@ -232,6 +232,42 @@ def load(cls, *args, **kwargs):
...
+
+    :param phrases: Class Phrases or Phraser
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: should be a link
```
 :class:`~gensim.models.phrases.Phrases` or :class:`~gensim.models.phrases.Phraser` 
```

File: gensim/models/phrases.py [hunk 5]
```diff
@@ -232,6 +232,42 @@ def load(cls, *args, **kwargs):
...
+    :param sentence: sentence to be processed
+    :returns: list of tokens from the sentence, where phrases ae joined with the delimeter."""
+
+
+    is_single, sentence = _is_single(sentence)
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: PEP8: too many blank lines before

File: gensim/models/phrases.py [hunk 6]
```diff
@@ -597,33 +633,8 @@ def __getitem__(self, sentence):
...
-                min_count=float(self.min_count),
-                corpus_word_count=float(self.corpus_word_count),
-            ),
-        )
-        new_s = []
-        for words, score in bigrams:
-            if score is not None:
-                words = delimiter.join(words)
-            new_s.append(words)
+        return sentence2token(self, sentence)
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: PEP8: too many blank lines after

File: gensim/models/phrases.py [hunk 7]
```diff
@@ -855,24 +866,8 @@ def __getitem__(self, sentence):
...
-            threshold=self.threshold,
-            common_terms=self.common_terms,
-            scorer=None)  # we will use our score_item function redefinition
-        new_s = []
-        for words, score in bigrams:
-            if score is not None:
-                words = delimiter.join(words)
-            new_s.append(words)
-        return [utils.to_unicode(w) for w in new_s]
+
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: PEP8: blank line contains whitespace

**Comment by jenishah** [comment]
Thanks for the review @menshikh-iv . This was my first PR, and I need to learn a lot about proper documentation. Your reviews were a great help.

File: gensim/models/phrases.py [hunk 8]
```diff
@@ -232,6 +232,51 @@ def load(cls, *args, **kwargs):
         return model
 
 
+def _sentence2token(phrase_class, sentence):
+    """ returns tokens after from a sentence by joining the phrases with delimiter.
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: Should be in form "Do X", not "does X"

File: gensim/models/phrases.py [hunk 9]
```diff
@@ -232,6 +232,51 @@ def load(cls, *args, **kwargs):
...
+    Parameters
+    ----------
+    phrase_class :
+        class:`~gensim.models.phrases.Phrases` or :class:`~gensim.models.phrases.Phraser`
+    sentence : {list of str, iterable of list of str}
+            Sentence or text corpus.
+
+    Returns
+    -------
+    {list of str, :class:`gensim.interfaces.TransformedCorpus`}
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: missed `~` for `TransformedCorpus` (as in other examples)

File: gensim/models/phrases.py [hunk 10]
```diff
@@ -232,6 +232,51 @@ def load(cls, *args, **kwargs):
...
+        `sentence` with detected phrase bigrams merged together, or a streamed corpus of such sentences
+        if the input was a corpus.
+    """
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: Empty line at the end of docstring

**Comment by menshikh-iv** [comment]
Nice work, thanks @jenishah, congratz with first contribution and happy hacktoberfest :star2:

**Pull request merged**